x/leveragelp/keeper/query_position.go

```diff
@@ -21,12 +23,31 @@ func (k Keeper) Position(goCtx context.Context, req *types.PositionRequest) (*ty
 	if err != nil {
 		return nil, err
 	}
+	updatedLeveragePosition := types.QueryPosition{}
+
+	pool, found := k.amm.GetPool(ctx, position.AmmPoolId)
+	if !found {
+		return nil, errorsmod.Wrap(ammtypes.ErrPoolNotFound, fmt.Sprintf("poolId: %d", position.AmmPoolId))
+	}
+	lp_price, err := pool.LpTokenPrice(ctx, k.oracleKeeper)
+	if err != nil {
+		return nil, err
+	}
+
+	lp_usd_price := position.LeveragedLpAmount.Mul(lp_price.TruncateInt())
+	price := k.oracleKeeper.GetAssetPriceFromDenom(ctx, position.Collateral.Denom)
+	updated_leverage := lp_usd_price.Quo(lp_usd_price.Sub(position.Liabilities.Mul(price.TruncateInt())))
+	
+	updatedLeveragePosition = types.QueryPosition{
+		Position:        &position,
+		UpdatedLeverage: updated_leverage,
+	}
```

Comment:
I see we are repeating this code logic, 3 times.
Can you reuse and add a function for this ?